File: informatika ismeretek/kozep/2019_okt/f#/Cbradio.fs
open System
open System.IO

let bejegyzestKeszit (k: string[]) = {| Ora = int k.[0]; Perc = int k.[1]; Adasok = int k.[2]; Nev = k.[3]|}
let bejegyzesek = File.ReadLines("cb.txt") |> Seq.skip(1)
                                           |> Seq.map(fun k -> k.Split(';') |> bejegyzestKeszit)
                                           |> Seq.toArray

printfn "3. Feladat: Bejegyzések száma: %d" bejegyzesek.Length

let voltEEloadas = if bejegyzesek |> Seq.exists(fun k -> k.Adasok = 4) then "4. Feladat: Volt 4 adást indító sofőr"
                                                                       else "4. Feladat: Nem volt 4 adást indító sofőr"
printfn "%s" voltEEloadas
printfn "5. Feladat: Írj be egy nevet!"

let bekertNev = Console.ReadLine()
let bekertHasznalatok = bejegyzesek |> Seq.filter(fun k -> k.Nev = bekertNev)
                                    |> Seq.sumBy(fun k -> k.Adasok)

if bekertHasznalatok > 0 then
    printfn "%s %dx használta a rádiót" bekertNev bekertHasznalatok
else
    printfn "Nincs ilyen nevű sofőr!"

let atszamolPercre (ora: int, perc: int) = ora * 60 + perc
let fileContent = bejegyzesek |> Seq.map(fun k -> string(atszamolPercre(k.Ora, k.Perc)) + ";" + k.Nev + ";" + string(k.Adasok))
                              |> Seq.toArray

File.WriteAllText("cb2.txt", "Kezdes;Nev;AdasDb\n" + String.Join("\n", fileContent))

bejegyzesek |> Seq.map(fun k -> k.Nev)
            |> Seq.distinct
            |> Seq.length
            |> printfn "8. Feladat: Sofőrök száma: %d"

bejegyzesek |> Seq.groupBy(fun k -> k.Nev)
            |> Seq.map(fun (k, v) -> {| Key = k; Value = v |> Seq.sumBy(fun m -> m.Adasok) |})
            |> Seq.maxBy(fun k -> k.Value)
            |> fun k -> printfn "9. Feladat: Legtöbb adás sofőre: %s: %d db" k.Key k.Value

File: informatika_ismeretek/kozep/2019_okt/python/Cbradio.py
class Bejegyzes:

    def __init__(self, line: str):
        split = line.split(';')

        self.ora = int(split[0])
        self.perc = int(split[1])
        self.adasok = int(split[2])
        self.nev = split[3].strip()

def atszamolPercre(ora: int, perc: int):
    return ora * 60 + perc


with open('cb.txt', 'r') as file:
    bejegyzesek = [ Bejegyzes(k) for k in file.readlines()[1:] ]

print(f'3. Feladat: Bejegyzések száma: {len(bejegyzesek)}')

voltE4Adasos = any(True for k in bejegyzesek if k.adasok == 4)
print(f'4. Feladat: {"Volt" if voltE4Adasos else "Nem volt"} 4 adást indító sofőr')

bekertNev = input('5. Feladat: Írj be egy nevet! ')
bekertHasznalatok = sum(k.adasok for k in bejegyzesek if k.nev == bekertNev)

if bekertHasznalatok > 0:
    print(f'{bekertNev} {bekertHasznalatok}x használta a rádiót')
else:
    print('Nincs ilyen nevű sofőr!')

with open('cb2.txt', 'w') as file:
    file.write('Kezdes;Nev;AdasDb\n')

    for bejegyzes in bejegyzesek:
        file.write(f'{atszamolPercre(bejegyzes.ora, bejegyzes.perc)};{bejegyzes.nev};{bejegyzes.adasok}\n')

soforokAdasszamokkal = {}
for bejegyzes in bejegyzesek:
    soforNeve = bejegyzes.nev

    soforokAdasszamokkal[soforNeve] = soforokAdasszamokkal.get(soforNeve, 0) + bejegyzes.adasok

legtobbAdasBejegyzes = max((k for k in soforokAdasszamokkal.items()), key = lambda k: k[1])

print(f'9. Feladat: Legtöbb adást indító sofőr: {legtobbAdasBejegyzes[0]}, adások: {legtobbAdasBejegyzes[1]}')